src/actions/measurements.js

```diff
@@ -0,0 +1,241 @@
+    "measurementsGroupBy",
+    "measurementsDisplay",
+    "measurementsShowOverallMean",
+    "measurementsShowThreshold",
+    "measurementsFilters"
+  ];
+  // Copy current control options for measurements
+  const newControls = pick(controls, measurementsControls);
+  // Checks the current group by is available as a field in collection
+  if (!collection.fields.has(newControls.measurementsGroupBy)) {
+    // If current group by is not available as a field, then default to the first grouping option.
+    newControls.measurementsGroupBy = collection.groupings[0];
+  }
+
+  // Verify that current filters are valid for the new collection
+  Object.entries(newControls.measurementsFilters).forEach(([field, valuesMap]) => {
+    // Delete filter for field that does not exist in the new collection filters
+    if (!collection.filters.has(field)) {
+      return delete newControls.measurementsFilters[field];
+    }
+    // Clone nested Map to avoid changing redux state in place
+    newControls.measurementsFilters[field] = new Map(valuesMap);
+    return [...valuesMap.keys()].forEach((value) => {
+      // Delete filter for values that do not exist within the field of the new collection
+      if (!collection.filters.get(field).values.has(value)) {
+        newControls.measurementsFilters[field].delete(value);
+      }
+    });
+  });
+
+  if (collection["display_defaults"]) {
+    const {
+      group_by,
+      measurement_display,
+      show_overall_mean,
+      show_threshold
+    } = collection["display_defaults"];
+
+    if (group_by) {
+      newControls.measurementsGroupBy = group_by;
+    }
+    if (measurement_display) {
+      newControls.measurementsDisplay = measurement_display;
+    }
+    if (typeof show_overall_mean === "boolean") {
+      newControls.measurementsShowOverallMean = show_overall_mean;
+    }
+    if (typeof show_threshold === "boolean") {
+      newControls.measurementsShowThreshold = show_threshold;
+    }
+  }
+
+  return newControls;
+};
+
+export const loadMeasurements = (json) => (dispatch, getState) => {
+  // TODO: Load controls from state to get potential url query parameters
+  const { tree, controls } = getState();
+  if (!tree.loaded) {
+    throw new Error("tree not loaded");
+  }
+
+  const collections = json["collections"];
+  if (!collections || collections.length === 0) {
+    throw new Error("Measurements JSON does not have collections");
+  }
+
+  collections.forEach((collection) => {
+    /*
+     * Create fields Map for easier access of titles and to keep ordering
+     * First add fields from JSON to keep user's ordering
+     * Then loop over measurements to add any remaining fields
+     */
+    const collectionFieldsArray = collection.fields;
+    collection.fields = new Map();
+    if (collectionFieldsArray && collectionFieldsArray.length > 0) {
+      collectionFieldsArray.forEach(({key, title}) => {
+        collection.fields.set(key, {title: title || key});
+      });
+    }
```

Comment:
I think something like the following is, at a glance, a bit more clear about what's going on here, but YMMV.

```js
    collection.fields = new Map(
      (collection.fields || [])
        .map(({key, title}) => [key, {title: title || key}])
    );
```